modules/openapi-generator/src/main/resources/go-server/controller-api.mustache

```diff
@@ -90,16 +90,21 @@ func (c *{{classname}}Controller) {{nickname}}(w http.ResponseWriter, r *http.Re
 	}
 	{{/isInteger}}
 	{{#isBoolean}}
-	{{paramName}}, err := parseBoolParameter(query.Get("{{baseName}}"))
+	{{paramName}}, err := parseBoolParameter(query.Get("{{baseName}}"), {{required}})
 	if err != nil {
-		w.WriteHeader(500)
+		w.WriteHeader(http.StatusBadRequest)
 		return
```

Comment:
I would suggest creating another error type, see RequiredError in `error.go` (see newer commit).

Since the `errors.New(errMsgRequiredMissing)` is now a global var (see other comment).
If the err = `errors.New(errMsgRequiredMissing)` var; then;
c.errorHandler(w, r, &YourNewlyCreatedError{Field: {{baseName}}}, nil)
else
c.errorHandler(w, r, err, nil)

Reply:
The issue with this is that any err, even non `errors.New(errMsgRequiredMissing)` would return StatusBadRequest